tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsManager.cs

```diff
@@ -24,28 +27,45 @@ internal class DataStreamsManager
     private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<DataStreamsManager>();
     private readonly NodeHashBase _nodeHashBase;
     private bool _isEnabled;
+    private DataStreamsWriter? _writer;
```

Comment:
What do you think of having a `IDataStreamsWriter` and introducing a `NoOpDataStreamsWriter` to replace the null field? Would that be overkill?

Reply:
I noticed below that adding a stats point involves `writer?.Add(...)`. If we went with a NoOp pattern then we wouldn't want to create the stats point for the no-op case, so we'd want to introduce a property/field `IDataStreamsWriter.Enabled` to branch on. Maybe that would be overkill..

Reply:
I'm fine with a NoOp writer if you prefer that. I don't think the `IDataStreamsWriter.Enabled` field is worth the hassle though - the `StatsPoint` is a `readonly struct` so shouldn't have much impact + we only replace the NoOp writer on app shutdown, so don't expect it to be called much.

Reply:
> It does make me wonder whether [this PR](https://github.com/DataDog/dd-trace-dotnet/pull) is correct though

Looks like the link got messed up, which PR did you mean?

Reply:
doh, I meant this one: https://github.com/DataDog/dd-trace-dotnet/pull/3220, looks like you spotted the issue there 😄

Reply:
Rather than cause cascading rebase issues here, I'll make the switch to `NoopDataStreamsWriter` in https://github.com/DataDog/dd-trace-dotnet/pull/3220, and also address the stats points issue there